presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecution.java

```diff
@@ -418,16 +429,16 @@ public void splitsAdded(int count, long weightSum)
         private boolean closed;
 
         @Nullable
-        private final ScheduledSplit partitionedSplit;
+        private List<ScheduledSplit> scheduledSplits;
 
         @GuardedBy("this")
         private Driver driver;
 
-        private DriverSplitRunner(DriverSplitRunnerFactory driverSplitRunnerFactory, DriverContext driverContext, @Nullable ScheduledSplit partitionedSplit)
+        private DriverSplitRunner(DriverSplitRunnerFactory driverSplitRunnerFactory, DriverContext driverContext, @Nullable List<ScheduledSplit> scheduledSplits)
         {
             this.driverSplitRunnerFactory = requireNonNull(driverSplitRunnerFactory, "driverFactory is null");
             this.driverContext = requireNonNull(driverContext, "driverContext is null");
-            this.partitionedSplit = partitionedSplit;
+            this.scheduledSplits = scheduledSplits;
```

Comment:
do a `requireNonNull ` check for the `scheduledSplits `?

Reply:
this is a nullable field.